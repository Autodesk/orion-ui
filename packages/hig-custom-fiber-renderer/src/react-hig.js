/**
Copyright 2016 Autodesk,Inc.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

*/
import React, { PropTypes } from 'react';
import ReactDOM from 'react-dom';

import { createElement } from './react-hig-elements';
import prepareUpdate from './prepareUpdate';

export default function createComponent(type) {
  const Adapter = class extends React.Component {
    static HIG_COMPONENT = true;

    constructor(props) {
      super(props);
      this.instance = createElement(type, props);
    }

    componentDidMount() {
      this._el = ReactDOM.findDOMNode(this);
      this._mount = this._el.parentNode;

      this._anchor = document.createComment('anchor');

      if (!this._mount) {
        throw new Error('can only mount if there is a parentNode');
      }

      this._mount.replaceChild(this._anchor, this._el);

      if (!this.context.parent) {
        this.instance.mount(this._mount, this._anchor);
      } else {
        this.context.parent.appendChild(this.instance);
      }
    }

    // renderSlot(props) {
    //   if (type !== 'slot') {
    //     return;
    //   }

    //   const element = React.isValidElement(props.children)
    //     ? props.children
    //     : <div>{props.children}</div>;
    //   ReactDOM.unstable_renderSubtreeIntoContainer(
    //     this,
    //     element,
    //     this.instance.getSlotNode()
    //   );
    // }

    componentWillUnmount() {
      this._mount.replaceChild(this._el, this._anchor);
      this.instance.unmount();
    }

    componentWillReceiveProps(nextProps) {
      const updatePayload = prepareUpdate(this.props, nextProps);
      this.instance.commitUpdate(updatePayload, this.props, nextProps);
      // this.renderSlot(nextProps);
    }

    render() {
      React.Children.forEach(this.props.children, child => {
        if (child && child.type && !child.type.HIG_COMPONENT) {
          console.error(
            `HIG unapproved! ${type} can not render DOM elements directly. Tried to render ${child.type}.`
          );
        }
      });

      return <hig-component>{this.props.children}</hig-component>;
    }

    getChildContext() {
      return {
        parent: this.instance
      };
    }
  };

  Adapter.displayName = type;

  Adapter.contextTypes = {
    parent: PropTypes.shape({
      appendChild: PropTypes.func
    })
  };

  Adapter.childContextTypes = {
    parent: PropTypes.shape({
      appendChild: PropTypes.func
    })
  };

  return Adapter;
}

export const Button = createComponent('hig-button');
export const Menu = createComponent('hig-menu');
Menu.Top = createComponent('hig-menu-top');
Menu.Sidebar = createComponent('hig-sidebar');
Menu.Sidebar.Group = createComponent('hig-sidebar-group');
Menu.Sidebar.Item = createComponent('hig-sidebar-item');
