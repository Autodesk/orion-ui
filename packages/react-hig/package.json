{
  "name": "@orion-ui/react-hig",
  "version": "1.0.3",
  "description": "hig components in react",
  "module": "lib/react-hig.es.js",
  "main": "lib/react-hig.js",
  "umd:main": "lib/react-hig.umd.js",
  "css": "lib/react-hig.css",
  "author": "Autodesk Inc.",
  "homepage": ".",
  "license": "Apache-2.0",
  "peerDependencies": {
    "react": "^15.4.1",
    "react-dom": "^15.4.1",
    "prop-types": "^15.5.8"
  },
  "devDependencies": {
    "hig.web": "^1.0.0",
    "babel-plugin-object-rest-spread": "0.0.0",
    "babel-plugin-transform-class-properties": "^6.24.1",
    "babel-preset-es2015-rollup": "^3.0.0",
    "ionicons": "^2.0.1",
    "jest-enzyme": "2.1.2",
    "prop-types": "^15.5.8",
    "react": "^15.4.1",
    "react-devtools-core": "^2.0.12",
    "react-dom": "^15.4.1",
    "react-scripts": "0.9.5",
    "rollup": "^0.41.6",
    "rollup-plugin-babel": "^2.7.1",
    "rollup-plugin-commonjs": "^8.0.2",
    "rollup-plugin-css-only": "^0.2.0",
    "rollup-plugin-node-resolve": "^3.0.0",
    "rollup-watch": "^3.2.2"
  },
  "scripts": {
    "build": "npm run lib && react-scripts build",
    "lib": "rollup -c && cp node_modules/hig.web/dist/hig.css lib/react-hig.css",
    "playground": "react-scripts start",
    "start": "npm run lib -- --watch",
    "test": "react-scripts test --env=jsdom",
    "test-ci": "react-scripts test --env=jsdom --coverage --collectCoverageFrom=src/**/*.{ts,tsx}"
  }
}