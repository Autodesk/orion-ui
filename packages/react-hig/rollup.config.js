/**
Copyright 2016 Autodesk,Inc.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

*/
import babel from 'rollup-plugin-babel';
import fs from 'fs';
import resolve from 'rollup-plugin-node-resolve';
import commonjs from 'rollup-plugin-commonjs';
import es2015Rollup from 'babel-preset-es2015-rollup';
import reactPreset from 'babel-preset-react';

const pkg = JSON.parse(fs.readFileSync('./package.json'));

export default {
  entry: 'src/react-hig.js',
  plugins: [
    resolve(),

    commonjs({
      namedExports: {
        '../hig.web/dist/hig.js': ['Button', 'GlobalNav']
      }
    }),

    babel({
      exclude: 'node_modules/**',
      babelrc: false,
      presets: [es2015Rollup, reactPreset],
      plugins: [
        'transform-class-properties',
        [
          'transform-object-rest-spread',
          {
            useBuiltIns: true
          }
        ]
      ]
    })
  ],
  targets: [
    { dest: pkg.main, format: 'cjs' },
    { dest: pkg.module, format: 'es' },
    { dest: pkg['umd:main'], format: 'umd', moduleName: 'ReactHIG' }
  ],

  external: ['react', 'react-dom', 'prop-types'],
  globals: {
    react: 'React',
    ['react-dom']: 'ReactDOM',
    ['prop-types']: 'PropTypes'
  }
};
