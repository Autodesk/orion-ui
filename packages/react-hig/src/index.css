
html, body, #root {
  height: 100%;
  margin: 0;

}

.playground-content{
  background-color: #FFFFFF;
}

section {
  margin-bottom: 40px;
}