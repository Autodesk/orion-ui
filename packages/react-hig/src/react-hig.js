/**
Copyright 2016 Autodesk,Inc.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

*/
import * as PropTypes from 'prop-types';
import React from 'react';
import ReactDOM from 'react-dom';

import { createElement, types } from './react-hig-elements';
import prepareUpdate from './prepareUpdate';

function createComponent(type) {
  const Adapter = class extends React.Component {
    static HIG_COMPONENT = true;

    constructor(props, context) {
      super(props);

      if (context.parent) {
        this.instance = context.parent.createElement(type, props);
      } else {
        this.instance = createElement(type, props);
      }
    }

    componentDidMount() {
      this._el = ReactDOM.findDOMNode(this);
      this._mount = this._el.parentNode;

      this._anchor = document.createComment(`${type}-anchor`);

      if (!this._mount) {
        throw new Error('can only mount if there is a parentNode');
      }

      this._mount.replaceChild(this._anchor, this._el);

      if (!this.context.parent) {
        this.instance.mount(this._mount, this._anchor);
      } else {
        const insertBeforeIndex = Array.from(
          this._anchor.parentNode.childNodes
        ).indexOf(this._anchor);

        if (
          this._anchor.nextSibling &&
          this._anchor.nextSibling.nodeType ===
            this._anchor.nextSibling.COMMENT_NODE
        ) {
          this.context.parent.insertBefore(this.instance, insertBeforeIndex);
        } else {
          this.context.parent.appendChild(this.instance);
        }
      }
    }

    componentWillUnmount() {
      this._mount.replaceChild(this._el, this._anchor);

      if (this.context.parent) {
        this.context.parent.removeChild(this.instance);
      } else {
        this.instance.unmount();
      }
    }

    componentWillReceiveProps(nextProps) {
      const updatePayload = prepareUpdate(this.props, nextProps);

      if (updatePayload) {
        this.instance.commitUpdate(updatePayload, this.props, nextProps);
      }
    }

    render() {
      return React.createElement(type, {}, this.props.children);
    }

    getChildContext() {
      if (this.instance.appendChild) {
        return {
          parent: this.instance
        };
      } else {
        return {
          parent: null
        };
      }
    }
  };

  Adapter.displayName = type;

  Adapter.contextTypes = {
    parent: PropTypes.shape({
      appendChild: PropTypes.func
    })
  };

  Adapter.childContextTypes = {
    parent: PropTypes.shape({
      appendChild: PropTypes.func
    })
  };

  return Adapter;
}

function createSlotComponent() {
  const Adapter = class extends React.Component {
    componentDidMount() {
      if (this.context.parent) {
        this.context.parent.addSlot(this.container);
      }
    }

    render() {
      const { children, ...rest } = this.props;
      return <div ref={r => this.container = r} {...rest}>{children}</div>;
    }

    getChildContext() {
      return {
        parent: null
      };
    }
  };

  Adapter.displayName = `hig-slot`;

  // Reset all the context for any children of the slot
  // In case we go back to the HIG context
  Adapter.childContextTypes = {
    parent: PropTypes.shape({
      appendChild: PropTypes.func
    })
  };

  Adapter.contextTypes = {
    parent: PropTypes.shape({
      addSlot: PropTypes.func
    })
  };

  return Adapter;
}

export const Button = createComponent(types.BUTTON);
export const Slot = createSlotComponent();
export const GlobalNav = createComponent(types.GLOBAL_NAV);

GlobalNav.Container = createComponent(types.CONTAINER);
GlobalNav.Container.TopNav = createComponent(types.TOP_NAV);
GlobalNav.Container.SubNav = createComponent(types.SUB_NAV);
GlobalNav.SideNav = createComponent(types.SIDE_NAV);
GlobalNav.SideNav.Sections = createComponent(types.SIDE_NAV_SECTIONS);
GlobalNav.SideNav.Sections.Item = createComponent(types.SIDE_NAV_SECTION);
GlobalNav.SideNav.Links = createComponent(types.SIDE_NAV_LINKS);
GlobalNav.SideNav.Links.Item = createComponent(types.SIDE_NAV_LINK);
GlobalNav.SideNav.Sections.Group = createComponent(types.SIDE_NAV_GROUP);
GlobalNav.SideNav.Sections.Group.Item = createComponent(types.SIDE_NAV_ITEM);
