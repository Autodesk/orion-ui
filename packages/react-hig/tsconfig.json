{
  "compilerOptions": {
    "module": "commonjs",
    "target": "es6",
    "moduleResolution": "node",
    "checkJs": true,
    "allowJs": true,
    "noEmit": true,
    "pretty": true,
    "jsx": "preserve",
    "allowSyntheticDefaultImports": true
  },
  "include": [
    "src"
  ]
}