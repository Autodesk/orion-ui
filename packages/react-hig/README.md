# react-hig

React components for the HIG

## Table of Contents

-   [Install](#install)
-   [Usage](#usage)

This project uses [node](http://nodejs.org) and [npm](https://npmjs.com). Go check them out if you don't have them locally installed.

```sh
$ npm install --save @orion-ui/react-hig
```

Then with a module bundler like [rollup](http://rollupjs.org/) or [webpack](https://webpack.js.org/), use as you would anything else:

```javascript
// using ES6 modules
import { Button } from '@orion-ui/react-hig';

// using CommonJS modules
var ReactHIG = require('@orion-ui/react-hig');
var Button = ReactHIG.Button;
```

The [UMD](https://github.com/umdjs/umd) build is also available on [unpkg](https://unpkg.com):

```html
<script src="https://unpkg.com/@orion-ui/react-hig/lib/react-hig.umd.js"></script>
```

You can find the library on `window.ReactHIG`.

## Usage

```jsx
import React from 'react';
import ReactDOM from 'react-dom';
import { Button } from '@orion-ui/react-hig';

class App extends React.Component {
  render() {
    return (
      <div>
        <Button title="Hello World!" />
      </div>
    )
  }
}
```

