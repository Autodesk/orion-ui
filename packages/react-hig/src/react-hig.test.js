/**
Copyright 2016 Autodesk,Inc.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

*/
import { mount } from 'enzyme';

import * as HIG from 'hig.web';
import React from 'react';

import { Button, GlobalNav } from './react-hig';

describe('react-hig', () => {
  describe('<Button>', () => {
    /**
     * Creates a hig.web button and returns the instance and it's container
     *
     * @param {object} defaults hig.web defaults for the button
     */
    function createHigButton(defaults = {}) {
      const higContainer = document.createElement('div');

      // use spread here to clone defaults since HIG.Button mutates this object
      const higButton = new HIG.Button({ ...defaults });

      higButton.mount(higContainer);

      return { higButton, higContainer };
    }

    it('renders the standard button', () => {
      const defaults = { title: 'regular button', link: 'http://example.com' };

      const { higButton, higContainer } = createHigButton(defaults);

      const container = document.createElement('div');

      const wrapper = mount(<Button {...defaults} />, { attachTo: container });

      expect(container.firstChild.outerHTML).toMatchSnapshot();

      expect(container.firstChild.outerHTML).toEqual(
        higContainer.firstChild.outerHTML
      );
    });

    it('updates the title the same as setTitle', () => {
      const { higButton, higContainer } = createHigButton();

      const newTitle = 'correct title';

      // update via hig API
      higButton.setTitle(newTitle);

      const container = document.createElement('div');

      // update via React API
      const wrapper = mount(<Button />, { attachTo: container });

      wrapper.setProps({ title: newTitle });

      expect(container.firstChild.outerHTML).toMatchSnapshot();

      expect(container.firstChild.outerHTML).toEqual(
        higContainer.firstChild.outerHTML
      );
    });

    it('removes the title if it is no longer specified', () => {
      const defaults = { title: 'some title' };

      const { higButton, higContainer } = createHigButton(defaults);

      const reactContainer = document.createElement('div');

      const wrapper = mount(<Button {...defaults} />, {
        attachTo: reactContainer
      });

      higButton.setTitle(null);

      // update to remove the title prop
      wrapper.setProps({ title: undefined });

      expect(reactContainer.firstChild.outerHTML).toMatchSnapshot();

      expect(reactContainer.firstChild.outerHTML).toEqual(
        higContainer.firstChild.outerHTML
      );
    });

    it('updates the link the same as setLink', () => {
      const { higButton, higContainer } = createHigButton();
      const newLink = 'http://example.com/2';

      higButton.setLink(newLink);

      const reactContainer = document.createElement('div');

      const wrapper = mount(<Button />, { attachTo: reactContainer });

      wrapper.setProps({ link: newLink });

      expect(reactContainer.firstChild.outerHTML).toMatchSnapshot();

      expect(reactContainer.firstChild.outerHTML).toEqual(
        higContainer.firstChild.outerHTML
      );
    });

    ['onClick', 'onHover'].forEach(eventName => {
      it(`sets up ${eventName} initially`, () => {
        const eventSpy = jest.fn();

        const reactContainer = document.createElement('div');

        const wrapper = mount(<Button {...{ [eventName]: eventSpy }} />, {
          attachTo: reactContainer
        });

        // click on the rendered button
        const instance = wrapper.instance().instance;

        // This is the same fn we pass to hig.web
        instance.events[eventName]();

        // expect onClickSpy to be called
        expect(eventSpy).toBeCalled();
      });

      it(`updates ${eventName} when it updates`, () => {
        const eventSpy1 = jest.fn();
        const eventSpy2 = jest.fn();

        const reactContainer = document.createElement('div');

        const wrapper = mount(<Button {...{ [eventName]: eventSpy1 }} />, {
          attachTo: reactContainer
        });

        wrapper.setProps({ [eventName]: eventSpy2 });

        // click on the rendered button
        const instance = wrapper.instance().instance;

        // This is the same fn we pass to hig.web
        instance.events[eventName]();

        // expect onClickSpy to be called
        expect(eventSpy2).toBeCalled();
      });

      it(`updates ${eventName} when it wasn't specified to begin with`, () => {
        const eventSpy1 = jest.fn();

        const reactContainer = document.createElement('div');

        const wrapper = mount(<Button />, {
          attachTo: reactContainer
        });

        wrapper.setProps({ [eventName]: eventSpy1 });

        const instance = wrapper.instance().instance;

        // This is the same fn we pass to hig.web
        instance.events[eventName]();

        // expect onClickSpy to be called
        expect(eventSpy1).toBeCalled();
      });

      it(`removes ${eventName} if it is no longer specified`, () => {
        const eventSpy1 = jest.fn();

        const reactContainer = document.createElement('div');

        const wrapper = mount(<Button {...{ [eventName]: eventSpy1 }} />, {
          attachTo: reactContainer
        });

        wrapper.setProps({ [eventName]: undefined });

        const instance = wrapper.instance().instance;

        expect(instance.events[eventName]).toBeUndefined();
      });
    });
  });

  describe('<GlobalNav>', () => {
    /**
     * Creates a hig.web GlobalNav and returns the instance and it's container
     */
    function createHigNav() {
      const higContainer = document.createElement('div');

      // use spread here to clone defaults since HIG.Button mutates this object
      const higNav = new HIG.GlobalNav();

      higNav.mount(higContainer);

      return { higNav, higContainer };
    }

    it('renders the global nav', () => {
      const { higNav, higContainer } = createHigNav();

      const reactContainer = document.createElement('div');
      const wrapper = mount(<GlobalNav />, { attachTo: reactContainer });

      expect(reactContainer.firstChild.outerHTML).toMatchSnapshot();

      expect(reactContainer.firstChild.outerHTML).toEqual(
        higContainer.firstChild.outerHTML
      );
    });

    describe('sideNavOpen prop', () => {
      it('is equal to showSideNav', () => {
        const { higNav, higContainer } = createHigNav();
        const reactContainer = document.createElement('div');
        const wrapper = mount(<GlobalNav />, { attachTo: reactContainer });

        higNav.showSideNav();

        wrapper.setProps({ sideNavOpen: true });

        expect(reactContainer.firstChild.outerHTML).toMatchSnapshot();

        expect(reactContainer.firstChild.outerHTML).toEqual(
          higContainer.firstChild.outerHTML
        );

        higNav.hideSideNav();
        wrapper.setProps({ sideNavOpen: false });

        expect(reactContainer.firstChild.outerHTML).toMatchSnapshot();

        expect(reactContainer.firstChild.outerHTML).toEqual(
          higContainer.firstChild.outerHTML
        );
      });

      it('can showSideNav by default', () => {
        const { higNav, higContainer } = createHigNav();
        const reactContainer = document.createElement('div');
        const wrapper = mount(<GlobalNav sideNavOpen={true} />, {
          attachTo: reactContainer
        });

        higNav.showSideNav();

        expect(reactContainer.firstChild.outerHTML).toMatchSnapshot();

        expect(reactContainer.firstChild.outerHTML).toEqual(
          higContainer.firstChild.outerHTML
        );
      });
    });

    describe('<SideNav>', () => {
      it('can render the SideNav by default', () => {
        const { higNav, higContainer } = createHigNav();

        const sideNav = new higNav.partials.SideNav();
        higNav.addSideNav(sideNav);

        const reactContainer = document.createElement('div');
        const wrapper = mount(
          <GlobalNav>
            <GlobalNav.SideNav />
          </GlobalNav>,
          {
            attachTo: reactContainer
          }
        );

        expect(reactContainer.firstChild.outerHTML).toMatchSnapshot();

        expect(reactContainer.firstChild.outerHTML).toEqual(
          higContainer.firstChild.outerHTML
        );
      });

      it('can only render a single SideNav', () => {
        expect(() => {
          mount(
            <GlobalNav>
              <GlobalNav.SideNav />
              <GlobalNav.SideNav />
            </GlobalNav>
          );
        }).toThrowError(/only one SideNav is allowed/);
      });

      describe('<Sections>', () => {
        it('can render a <Sections> by default', () => {
          const { higNav, higContainer } = createHigNav();

          const sideNav = new higNav.partials.SideNav();
          higNav.addSideNav(sideNav);

          const reactContainer = document.createElement('div');
          const wrapper = mount(
            <GlobalNav>
              <GlobalNav.SideNav>
                <GlobalNav.SideNav.Sections />
              </GlobalNav.SideNav>
            </GlobalNav>,
            {
              attachTo: reactContainer
            }
          );

          expect(reactContainer.firstChild.outerHTML).toMatchSnapshot();

          expect(reactContainer.firstChild.outerHTML).toEqual(
            higContainer.firstChild.outerHTML
          );
        });

        it('can only render a single <Sections>', () => {
          expect(() => {
            mount(
              <GlobalNav>
                <GlobalNav.SideNav>
                  <GlobalNav.SideNav.Sections />
                  <GlobalNav.SideNav.Sections />
                </GlobalNav.SideNav>
              </GlobalNav>
            );
          }).toThrowError(/only one Sections is allowed/);
        });

        describe('<Section>', () => {
          it('can render a <Section> by default', () => {
            const { higNav, higContainer } = createHigNav();

            const sideNav = new higNav.partials.SideNav();
            higNav.addSideNav(sideNav);

            const sectionDefaults = {
              headerLabel: 'Project',
              headerName: 'Thunderstorm'
            };

            const section1 = new sideNav.partials.Section(sectionDefaults);

            sideNav.addSection(section1);

            const reactContainer = document.createElement('div');
            const wrapper = mount(
              <GlobalNav>
                <GlobalNav.SideNav>
                  <GlobalNav.SideNav.Sections>
                    <GlobalNav.SideNav.Sections.Item {...sectionDefaults} />
                  </GlobalNav.SideNav.Sections>

                </GlobalNav.SideNav>
              </GlobalNav>,
              {
                attachTo: reactContainer
              }
            );

            expect(reactContainer.firstChild.outerHTML).toMatchSnapshot();

            expect(reactContainer.firstChild.outerHTML).toEqual(
              higContainer.firstChild.outerHTML
            );
          });

          it('can render multiple sections', () => {
            const { higNav, higContainer } = createHigNav();

            const sideNav = new higNav.partials.SideNav();
            higNav.addSideNav(sideNav);

            const section1Defaults = {
              headerLabel: 'Project',
              headerName: 'Thunderstorm'
            };

            const section1 = new sideNav.partials.Section(section1Defaults);

            sideNav.addSection(section1);

            const section2Defaults = {
              headerLabel: 'Project',
              headerName: 'Boo Ya'
            };

            const section2 = new sideNav.partials.Section(section2Defaults);

            sideNav.addSection(section2);

            const reactContainer = document.createElement('div');
            const wrapper = mount(
              <GlobalNav>
                <GlobalNav.SideNav>
                  <GlobalNav.SideNav.Sections>
                    <GlobalNav.SideNav.Sections.Item {...section1Defaults} />
                    <GlobalNav.SideNav.Sections.Item {...section2Defaults} />
                  </GlobalNav.SideNav.Sections>
                </GlobalNav.SideNav>
              </GlobalNav>,
              {
                attachTo: reactContainer
              }
            );

            expect(reactContainer.firstChild.outerHTML).toMatchSnapshot();

            expect(reactContainer.firstChild.outerHTML).toEqual(
              higContainer.firstChild.outerHTML
            );
          });

          it('can update to have a section before the other section', () => {
            const { higNav, higContainer } = createHigNav();

            const sideNav = new higNav.partials.SideNav();
            higNav.addSideNav(sideNav);

            const section1Defaults = {
              headerLabel: 'Label 1',
              headerName: 'Name 1'
            };

            const section1 = new sideNav.partials.Section(section1Defaults);

            // DELIBERATELY DON'T ADD SECTION 1

            const section2Defaults = {
              headerLabel: 'Label 2',
              headerName: 'Name 2'
            };

            const section2 = new sideNav.partials.Section(section2Defaults);

            sideNav.addSection(section2);

            // ADD SECTION 1 before SECTION 2

            sideNav.addSection(section1, section2);

            class CustomComponent extends React.Component {
              constructor(props) {
                super(props);
                this.state = { showingItem: false };
              }

              render() {
                return (
                  <GlobalNav>
                    <GlobalNav.SideNav>
                      <GlobalNav.SideNav.Sections>
                        {this.state.showingItem &&
                          <GlobalNav.SideNav.Sections.Item
                            {...section1Defaults}
                          />}
                        <GlobalNav.SideNav.Sections.Item
                          {...section2Defaults}
                        />
                      </GlobalNav.SideNav.Sections>
                    </GlobalNav.SideNav>

                  </GlobalNav>
                );
              }
            }

            const reactContainer = document.createElement('div');
            const wrapper = mount(<CustomComponent />, {
              attachTo: reactContainer
            });

            expect(reactContainer.firstChild.outerHTML).toMatchSnapshot();

            wrapper.setState({ showingItem: true });

            expect(reactContainer.firstChild.outerHTML).toMatchSnapshot();

            expect(reactContainer.firstChild.outerHTML).toEqual(
              higContainer.firstChild.outerHTML
            );
          });

          it('can update the section headerLabel');

          it('can update the section headerName');
        });
      });
    });
  });
});
