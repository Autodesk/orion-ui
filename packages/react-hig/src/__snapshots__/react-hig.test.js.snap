exports[`react-hig <Button> removes the title if it is no longer specified 1`] = `"<a href=\"#\" class=\"hig__button\" title=\"some title\"></a>"`;

exports[`react-hig <Button> renders the standard button 1`] = `"<a href=\"http://example.com\" class=\"hig__button\" title=\"regular button\">regular button</a>"`;

exports[`react-hig <Button> updates the link the same as setLink 1`] = `"<a href=\"http://example.com/2\" class=\"hig__button\" title=\"link\">link</a>"`;

exports[`react-hig <Button> updates the title the same as setTitle 1`] = `"<a href=\"#\" class=\"hig__button\" title=\"link\">correct title</a>"`;

exports[`react-hig <GlobalNav> <SideNav> <Sections> <Section> can render a <Section> by default 1`] = `
"<div class=\"hig__global-nav\">
    <div class=\"hig__global-nav__sidenav\">
    <div class=\"hig__global-nav__side-nav__section\">
    <div class=\"hig__global-nav__side-nav__section__header\">
        <span class=\"hig__global-nav__side-nav__section__header__label\">Project</span>
        <span class=\"hig__global-nav__side-nav__section__header__name\">Thunderstorm</span>
    </div>
    <!--GROUP-->
</div><!--SECTION-->
    <div class=\"hig__global-nav__sidenav__links\">
        <!--LINK-->
    </div>
    <div class=\"hig__global-nav__sidenav__copywright\">© 2017 Autodesk, Inc. </div>
</div><!--SIDENAV-->
    <!--CONTAINER-->
</div>"
`;

exports[`react-hig <GlobalNav> <SideNav> <Sections> <Section> can render multiple sections 1`] = `
"<div class=\"hig__global-nav\">
    <div class=\"hig__global-nav__sidenav\">
    <div class=\"hig__global-nav__side-nav__section\">
    <div class=\"hig__global-nav__side-nav__section__header\">
        <span class=\"hig__global-nav__side-nav__section__header__label\">Project</span>
        <span class=\"hig__global-nav__side-nav__section__header__name\">Thunderstorm</span>
    </div>
    <!--GROUP-->
</div><div class=\"hig__global-nav__side-nav__section\">
    <div class=\"hig__global-nav__side-nav__section__header\">
        <span class=\"hig__global-nav__side-nav__section__header__label\">Project</span>
        <span class=\"hig__global-nav__side-nav__section__header__name\">Boo Ya</span>
    </div>
    <!--GROUP-->
</div><!--SECTION-->
    <div class=\"hig__global-nav__sidenav__links\">
        <!--LINK-->
    </div>
    <div class=\"hig__global-nav__sidenav__copywright\">© 2017 Autodesk, Inc. </div>
</div><!--SIDENAV-->
    <!--CONTAINER-->
</div>"
`;

exports[`react-hig <GlobalNav> <SideNav> <Sections> <Section> can update to have a section before the other section 1`] = `
"<div class=\"hig__global-nav\">
    <div class=\"hig__global-nav__sidenav\">
    <div class=\"hig__global-nav__side-nav__section\">
    <div class=\"hig__global-nav__side-nav__section__header\">
        <span class=\"hig__global-nav__side-nav__section__header__label\">Label 2</span>
        <span class=\"hig__global-nav__side-nav__section__header__name\">Name 2</span>
    </div>
    <!--GROUP-->
</div><!--SECTION-->
    <div class=\"hig__global-nav__sidenav__links\">
        <!--LINK-->
    </div>
    <div class=\"hig__global-nav__sidenav__copywright\">© 2017 Autodesk, Inc. </div>
</div><!--SIDENAV-->
    <!--CONTAINER-->
</div>"
`;

exports[`react-hig <GlobalNav> <SideNav> <Sections> <Section> can update to have a section before the other section 2`] = `
"<div class=\"hig__global-nav\">
    <div class=\"hig__global-nav__sidenav\">
    <div class=\"hig__global-nav__side-nav__section\">
    <div class=\"hig__global-nav__side-nav__section__header\">
        <span class=\"hig__global-nav__side-nav__section__header__label\">Label 1</span>
        <span class=\"hig__global-nav__side-nav__section__header__name\">Name 1</span>
    </div>
    <!--GROUP-->
</div><div class=\"hig__global-nav__side-nav__section\">
    <div class=\"hig__global-nav__side-nav__section__header\">
        <span class=\"hig__global-nav__side-nav__section__header__label\">Label 2</span>
        <span class=\"hig__global-nav__side-nav__section__header__name\">Name 2</span>
    </div>
    <!--GROUP-->
</div><!--SECTION-->
    <div class=\"hig__global-nav__sidenav__links\">
        <!--LINK-->
    </div>
    <div class=\"hig__global-nav__sidenav__copywright\">© 2017 Autodesk, Inc. </div>
</div><!--SIDENAV-->
    <!--CONTAINER-->
</div>"
`;

exports[`react-hig <GlobalNav> <SideNav> <Sections> can render a <Sections> by default 1`] = `
"<div class=\"hig__global-nav\">
    <div class=\"hig__global-nav__sidenav\">
    <!--SECTION-->
    <div class=\"hig__global-nav__sidenav__links\">
        <!--LINK-->
    </div>
    <div class=\"hig__global-nav__sidenav__copywright\">© 2017 Autodesk, Inc. </div>
</div><!--SIDENAV-->
    <!--CONTAINER-->
</div>"
`;

exports[`react-hig <GlobalNav> <SideNav> can render the SideNav by default 1`] = `
"<div class=\"hig__global-nav\">
    <div class=\"hig__global-nav__sidenav\">
    <!--SECTION-->
    <div class=\"hig__global-nav__sidenav__links\">
        <!--LINK-->
    </div>
    <div class=\"hig__global-nav__sidenav__copywright\">© 2017 Autodesk, Inc. </div>
</div><!--SIDENAV-->
    <!--CONTAINER-->
</div>"
`;

exports[`react-hig <GlobalNav> renders the global nav 1`] = `
"<div class=\"hig__global-nav\">
    <!--SIDENAV-->
    <!--CONTAINER-->
</div>"
`;

exports[`react-hig <GlobalNav> sideNavOpen prop can showSideNav by default 1`] = `
"<div class=\"hig__global-nav hig__global-nav--open\">
    <!--SIDENAV-->
    <!--CONTAINER-->
</div>"
`;

exports[`react-hig <GlobalNav> sideNavOpen prop is equal to showSideNav 1`] = `
"<div class=\"hig__global-nav hig__global-nav--open\">
    <!--SIDENAV-->
    <!--CONTAINER-->
</div>"
`;

exports[`react-hig <GlobalNav> sideNavOpen prop is equal to showSideNav 2`] = `
"<div class=\"hig__global-nav\">
    <!--SIDENAV-->
    <!--CONTAINER-->
</div>"
`;
